docs/sdk/v3/guides/liquidity/04-swap-and-add-liquidity.md

```diff
@@ -0,0 +1,97 @@
+```
+
+The second parameter, which is optional, is an instance of `SwapAndAddOptions`. If it is included, `routeToRatio` will return the calldata for executing the atomic swap-and-add. These options contain `swapConfig` and `addLiquidityOptions`. `swapConfig` configures to set a recipient of leftover dust from swap, `slippageTolerance` and a `deadline` for the swap.
+
+Then, `addLiquidityOptions` must contain a `tokenId` to add to an existing position, **or** a `recipient` to mint a new one. It also includes a `slippageTolerance` and `deadline` for adding liquidity.
+
+The only custom parameter for `swapAndAddOptions`, is the token id, which is no other than the position id of the last position we minted in the example:
+
+```js reference title="Constructing SwapAndAddOptions" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L48-L58
+```
+
+Thirdly, we construct two instances of `CurrencyAmount`s, each of which represents the initial balance of the token that we wish to swap-and-add, where the token is the token in our target liquidity pool. We use the configuration parameters of the guide to set those:
+
+```js reference title="Constructing the two CurrencyAmounts" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L60-L74
+```
+
+
+Finally we construct the position, a position object that contains the details of the position for which to add liquidity. The position liquidity can be set to 1, since liquidity is still unknown and will be set inside the call to `routeToRatio`:
+
+```js reference title="Making the call to routeToRatio" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L76-L79
+```
+
+
+### Making the swap-and function call
+
+Having constructed all the parameters we need to call `routeToRatio`, we can now make the call to the function:
+
+```js reference title="Constructing the position object" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L81-L87
+```
+The return type of the function call is [SwapToRatioResponse](https://github.com/Uniswap/smart-order-router/blob/97c1bb7cb64b22ebf3509acda8de60c0445cf250/src/routers/router.ts#L121). If a route was found successfully, this object will have two fields: the status will be set to success and the result will contain the `SwapToRatioRoute` object. We check to make sure that both of those conditions hold true before we construct and submit the transaction:
```

Comment:
```suggestion
The return type of the function call is [SwapToRatioResponse](https://github.com/Uniswap/smart-order-router/blob/97c1bb7cb64b22ebf3509acda8de60c0445cf250/src/routers/router.ts#L121). If a route was found successfully, this object will have two fields: the status (success) and the `SwapToRatioRoute` object. We check to make sure that both of those conditions hold true before we construct and submit the transaction:
```

What about if it fails? What will be different about the conditions in this case?

Reply:
For our example, if it fails, we deem the transaction as failed